crates/env/src/engine/off_chain/impls.rs

```diff
@@ -180,9 +182,53 @@ impl EnvInstance {
     {
         let mut full_scope: [u8; BUFFER_SIZE] = [0; BUFFER_SIZE];
         let full_scope = &mut &mut full_scope[..];
-        ext_fn(&self.engine, full_scope);
+        ext_fn(&self.engine.borrow_mut(), full_scope);
         scale::Decode::decode(&mut &full_scope[..]).map_err(Into::into)
     }
+
+    fn generate_address(
```

Comment:
maybe we should not leave this in the scope of impl envInstanse because it is not really uses EnvInstance